Sette_e_mezzo.py

```diff
@@ -26,91 +25,122 @@
-        if my_player and len(my_player.deck.cards):
-                Game.vistext(CardText,my_player.deck.cards[-1].name + " " + my_player.deck.cards[-1].value,BLACK)
-
-
-        # score
-        if my_player:
-                if my_player.score <= 7.5:
-                        scorecolor = GREEN
-                else:
-                        scorecolor = RED
-
-                Game.vistext(ScoreText,"Score: " + str(my_player.score if my_player else 0), scorecolor)
-                if my_player.score > 7.5:
-                        Game.vistext(Text((250,300),StdFont,20,scorecolor),"You lost! Score over 7.5!",)
-                        tmp_count = 0
-                        for card in my_player.deck.cards:
-                                Game.vissurf(card.img,((60*tmp_count),50))
-                                Game.vistext(Text(((18 + 60*tmp_count),160),StdFont,20,scorecolor,GRAY),card.code)
-                                tmp_count = tmp_count + 1
-                        
 
-        tmp_count = 0
-        # cards and codes
-        if gamestate and gamestate.isrunning:
-                if my_player:
-                        for card in my_player.deck.cards:
-                                Game.vissurf(card.img,((60*tmp_count),50))
-                                Game.vistext(Text(((18 + 60*tmp_count),160),StdFont,20,scorecolor,GRAY),card.code)
-                                tmp_count = tmp_count + 1
+        # Main game loop
+        while gamestate.isrunning:
 
+                # User input
+                for event in Game.events():
+                        if event.type == QUIT:
+                                Game.quit()
+                                sys.exit()
+                        if DealButton.check(event) == True and gamestate and gamestate.isrunning:
+                                gamestate.dealCards()
+                        if ResetButton.check(event) == True and gamestate and gamestate.isrunning:
+                                gamestate.endGame()
 
-                # remaining deck
-                if my_player:
-                        Game.vissurf(gamestate.dealer.deck.img, DeckText.xy)
-                        Game.vistext(DeckText,"Remaining deck = " + str(len(gamestate.dealer.deck.cards)), BLACK)
+                # Render background               
+                Game.display.fill(WHITE)
 
                         
-                        # keep score
-                        gamestate.updateScore()
-                        if my_player and my_player.score > 7.5:
-                                print("Score over 7.5, ending game!")
-                                gamestate.endGame()
+                # --print the cards and info to the display surface--
 
+                # latest card name and value
+                if my_player and len(my_player.deck.cards):
+                        Game.vistext(CardText,my_player.deck.cards[-1].name + " " + my_player.deck.cards[-1].value)
```

Comment:
`my_player.deck.cards[-1]` you're trying to access an element of a list with a value of `-1` this should cause an exception and won't work.

Reply:
Python allows negative indexing so [0,1,2,3,4,5,6] is also [-7,-5,-4,-3,-2,-1]

Reply:
What even. TIL